tensorflow/go/op/gradients.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2016 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package op
+
+import tf "github.com/tensorflow/tensorflow/tensorflow/go"
+
+// Gradients adds gradients computation ops to the graph according to scope.
+//
+// Arguments:
+//  y: output of the function to derive
+//  x: inputs of the function for which partial derivatives are computed
+//  dx: if not null, the partial derivatives of some loss function L w.r.t. y
+//
+//  return the partial derivatives
+func Gradients(scope *Scope, y []tf.Output, x []tf.Output, dx ...tf.Output) (output []tf.Output) {
+	var err error
+	if output, err = scope.graph.AddGradients(scope.opName(scope.uniqueName("Gradients")), y, x, dx); err != nil {
```

Comment:
I'm not sure we want to use `scope.uniqueName` here, just `scope.opName("Gradients")`.
The reason I say so is that we don't use this for any other ops (https://github.com/tensorflow/tensorflow/blob/e45c90f0e4d17ac22048a73f1e81bd9c7a7a5145/tensorflow/go/op/scope.go#L85 etc.), so I wouldn't do so here either - to be consistent with how primitive operations are added to the graph.

Reply:
Fixed.